app/models/az322_contribution.rb

```diff
@@ -29,6 +29,7 @@ class Az322Contribution < ApplicationRecord
   validates :ctds_code, presence: true, format: { with: /\A\d{9}\z/, message: -> (_object, _data) { I18n.t("validators.ctds_code") }}, if: -> { made_contribution == "yes" }
   validates :district_name, presence: true, if: -> { made_contribution == "yes" }
   validates :amount, presence: true, numericality: { greater_than: 0 }, if: -> { made_contribution == "yes" }
+  validate :amount_format
```

Comment:
Changing these two lines to:

```ruby
  validates :amount, presence: {message: 'must be a valid dollar amount'}, numericality: { greater_than: 0 }
  # validate :amount_format
```
Makes most tests pass. There are a couple of failures.

[One test fails](https://github.com/codeforamerica/vita-min/blob/be299371d99c8d4d258d3198b312150da3e629c6/spec/models/az322_contribution_spec.rb#L89) because the error message has changed (and is now "must be a valid dollar amount"), which is fine, imo.

[Another test fails](https://github.com/codeforamerica/vita-min/blob/be299371d99c8d4d258d3198b312150da3e629c6/spec/models/az322_contribution_spec.rb#L127-L131) because it rounds to the nearest hundreth, which I think is probably also fine.

[Notably, the test with the trailing decimal](https://github.com/codeforamerica/vita-min/blob/be299371d99c8d4d258d3198b312150da3e629c6/spec/models/az322_contribution_spec.rb#L121-L125) does _not_ fail, instead becoming value `1.0` instead.

Reply:
i'm not quite understanding your suggestion, @powersurge360? are you saying that the `VALID_MONEY_REGEXP` does the same thing as `numericality: { greater_than: 0 }`? even if it does i think it still should incorporate `read_attribute_before_type_cast` which i can't figure out how to do without adding a validator method...

Reply:
Yea, wanting to truncate vs allowing it to round in the way the `BigDecimal` does naturally is for sure the trickiest bit. The typecasting goes a long way towards making sure we're receiving valid numbers but if we're deviating things get wonky. Hrrrrrrm.

Reply:
>[Notably, the test with the trailing decimal](https://github.com/codeforamerica/vita-min/blob/be299371d99c8d4d258d3198b312150da3e629c6/spec/models/az322_contribution_spec.rb#L121-L125) does not fail, instead becoming value 1.0 instead.

this is b/c we take away the trailing decimal here:
>  def amount=(value)
    if value&.to_s&.match?(/^(\d+)?\.$/) # Remove trailing decimal. Ex: '10.'
      value = value.chop
    end
    write_attribute :amount, value
  end

outside of the validation. hope that clears things up!

Reply:
Speaking of, I see that's not part of this but you might consider dropping the regex and just doing `value = value.chomp('.')`. That will remove the trailing decimal if and only if it exists, otherwise returning the string unmodified. Or maybe as with side effects, if you prefer.

`value.chomp!('.')`

Reply:
ooh yes I had initially used this before but ran into issues but looks like I was just using this completely wrong

Reply:
It modifies the value in place. What it returns is just whether it changed and what it changed to, the actual value changes in place without needing to reassign the variable.